libs/hpc-ui/src/lib/components/form-fields/date-picker.tsx

```diff
@@ -1,46 +1,75 @@
+import { useEffect, useState, useCallback, useMemo } from 'react';
 import { useField } from 'formik';
 import { TextField, Link } from '@mui/material';
 import { DatePicker as BaseDatePicker } from '@mui/x-date-pickers/DatePicker';
 import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
 import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 import tw from 'twin.macro';
-import dayjs from '../../i18n/utils/dayjs';
-import { useEffect, useState } from 'react';
+import dayjs, { Dayjs } from 'dayjs';
 
 const StyledDatePicker = tw.div`
-flex
-w-full
-items-baseline
+  flex
+  w-full
+  items-baseline
 `;
 
 const StyledLink = tw(Link)`
-ms-8
+  ms-8
 `;
 
+interface DatePickerProps {
+  name: string;
+  label: string;
+  lang?: string;
+  enableButton?: boolean;
+  [key: string]: any; // Additional props
+}
+
```

Comment:
```suggestion
type DatePickerProps = {
  name: string;
  label: string;
  lang?: string;
  enableButton?: boolean;
} & BaseDatePickerProps<DayJS>;

```

Like this we can add props that `BaseDatePicker` can accept and allow us to modify things when we need it. (This comment makes reference to the comment of `checkbox.tsx`)